Lib/test/test_capi.py

```diff
@@ -416,6 +416,70 @@ class SubDict(dict):
         self.assertTrue(_testcapi.mapping_has_key(dct2, 'a'))
         self.assertFalse(_testcapi.mapping_has_key(dct2, 'b'))
 
+    def test_sequence_set_slice(self):
+        # Correct case:
+        data = [1, 2, 3, 4, 5]
+        data_copy = data.copy()
+
+        _testcapi.sequence_set_slice(data, 1, 3, [8, 9])
+        data_copy[1:3] = [8, 9]
+        self.assertEqual(data, data_copy)
+        self.assertEqual(data, [1, 8, 9, 4, 5])
+
+        # Custom class:
+        class Custom:
+            def __setitem__(self, index, value):
+                self.index = index
+                self.value = value
+
+        c = Custom()
+        _testcapi.sequence_set_slice(c, 0, 5, 'abc')
+        self.assertEqual(c.index, slice(0, 5))
+        self.assertEqual(c.value, 'abc')
+
+        # Immutable sequences must raise:
```

Comment:
Here it would be good to check that failed slices didn't change the sequence.

Reply:
Good idea! Done.

Thank you for all the reviews 🎉